src/server/events/index.ts

```diff
@@ -0,0 +1,114 @@
+import ical, { CalendarResponse, CalendarComponent, VEvent } from 'node-ical';
+import { AuthenticationError, UserInputError } from 'apollo-server-express';
+import { ObjectID } from 'mongodb';
+import { EventUpdateInput, EventDbObject } from '../generated/graphql';
+import { Models } from '../models';
+import { EventUpdate } from '../../client/routes/events/ManageEventTypes';
+
+const EVENTTYPE = 'VEVENT';
+
+const filterByCalType = (objNames: string[], cal: CalendarResponse, type: string): string[] => {
+	return objNames.filter(obj => cal[obj].type === type);
+};
+
+const filterCalByObjectNames = (objNames: string[], cal: CalendarResponse): CalendarComponent[] => {
+	return objNames.map(key => cal[key]);
+};
+
+// Helper func to simplify VEvent type because it was overly obtuse and dramatically hindered mocks for testing
+const extractVEventIntoRecord = (event: VEvent): Record<string, string> => {
+	return {
+		summary: event.summary,
+		start: new Date(event.start as Date).toUTCString(),
+		end: new Date(event.end as Date).toUTCString(),
+		location: event.location,
+		description: event.description,
+	};
+};
+
+export const transformCalEventToDBUpdate = (event: Record<string, string>): EventUpdate => {
+	if (!event.start || !event.end)
+		throw new AuthenticationError('Calendar event did not contain start or end timestamp');
+	const parsedStart = new Date(event.start);
+	const parsedEnd = new Date(event.end);
+	const parsedType = /\[(.*?)\]/.exec(event.summary); // Looks for a **single** [Event Type] tag in name
+	return {
+		name: event.summary.replace(/\s*\[(.*?)\]\s*/, ''),
```

Comment:
what does this regex do? add a comment?

Reply:
Removes all instances of "[<string here>]" with whitespace outside of brackets, comment added.